packages/components/organisms/f-checkout/src/services/CheckoutServiceApi.js

```diff
@@ -0,0 +1,51 @@
+import axios from 'axios';
+import { VUEX_CHECKOUT_EXPERIMENTATION_MODULE } from '../constants';
+
+export default {
+    async getCheckout (url, state, timeout) {
+        const authHeader = state.authToken && `Bearer ${state.authToken}`;
+        const config = {
+            headers: {
+                'Content-Type': 'application/json',
+                ...(state.authToken && {
+                    Authorization: authHeader
+                })
+            },
+            timeout
+        };
+
+        return axios.get(url, config);
+    },
+
+
+    async updateCheckout (url, state, rootGetters, data, timeout) {
+        const authHeader = state.authToken && `Bearer ${state.authToken}`;
+        const experimentationHeaders = rootGetters[`${VUEX_CHECKOUT_EXPERIMENTATION_MODULE}/getExperimentsHeaders`];
+        const config = {
+            headers: {
+                'Content-Type': 'application/json',
+                ...(state.authToken && {
+                    Authorization: authHeader
+                }),
+                ...experimentationHeaders
+            },
+            timeout
+        };
+
+        return axios.patch(url, data, config);
+    },
+
+    async getAvailableFulfilment (url, timeout) {
+        const config = {
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            timeout
+        };
+
+        return axios.get(url, config);
+    }
+};
+
+
+
```

Comment:
Nit: unnecessary empty lines :)

Reply:
Ah yep thank youu - I'll update and add more tests. Not sure about how to avoid the repetition with the config object but will look into that now

Reply:
We can take a look at the code duplication on a separate PR if needed.

Reply:
If it helps, on menu we had another external service method that accepts some parameters and conditionally adds the values to a config object e.g.

```js
const createHeaders = (headers, authToken) => {
    const authHeader = authToken && `Bearer ${authToken}`;

    return {
		'Content-Type': 'application/json',
        ...headers,
        ...(authHeader && {
            authorization: authHeader
        })
    };
};

...

const experimentationHeaders = rootGetters[`${VUEX_CHECKOUT_EXPERIMENTATION_MODULE}/getExperimentsHeaders`];

const headers = {
	'x-je-application-id': 7, // Responsive Web
    'x-je-application-version': applicationVersion,
	...experimentationHeaders
};

service.createHeaders(headers, authToken);
```

Reply:
This is awesome thank you! I'll see if we can do this too ⭐

Reply:
I'm gonna make a ticket to look at this in a separate PR ^ :)

Reply:
spoken to @charliearlie about this - a ticket will be added to the checkout board to get this done at some point 👍